cartography/intel/gcp/crm/folders.py

```diff
@@ -37,68 +58,55 @@ def load_gcp_folders(
+    :param org_resource_name: Full organization resource name (e.g., "organizations/123456789012")
     """
+    # Transform data in place: add parent_org or parent_folder fields based on parent type
     for folder in data:
+        folder["parent_org"] = None
+        folder["parent_folder"] = None
+
         if folder["parent"].startswith("organizations"):
-            query = "MATCH (parent:GCPOrganization{id:$ParentId})"
+            folder["parent_org"] = folder["parent"]
         elif folder["parent"].startswith("folders"):
-            query = """
-            MERGE (parent:GCPFolder{id:$ParentId})
-            ON CREATE SET parent.firstseen = timestamp()
-            """
+            folder["parent_folder"] = folder["parent"]
         else:
-            # Skip folders with unexpected parent types
             logger.warning(
-                f"Skipping folder {folder['name']} with unexpected parent type: {folder['parent']}"
+                f"Folder {folder['name']} has unexpected parent type: {folder['parent']}"
             )
-            continue
-        query += """
-        MERGE (folder:GCPFolder{id:$FolderName})
-        ON CREATE SET folder.firstseen = timestamp()
-        SET folder.foldername = $FolderName,
-            folder.displayname = $DisplayName,
-            folder.lifecyclestate = $LifecycleState,
-            folder.lastupdated = $gcp_update_tag
-        WITH parent, folder
-        MERGE (parent)-[r:RESOURCE]->(folder)
-        ON CREATE SET r.firstseen = timestamp()
-        SET r.lastupdated = $gcp_update_tag
-        """
-        neo4j_session.run(
-            query,
-            ParentId=folder["parent"],
-            FolderName=folder["name"],
-            DisplayName=folder.get("displayName", None),
-            LifecycleState=folder.get("lifecycleState", None),
-            gcp_update_tag=gcp_update_tag,
-        )
+            # Still include it but with both parent fields as None
 
-
-@timeit
-def cleanup_gcp_folders(
-    neo4j_session: neo4j.Session,
-    common_job_parameters: Dict,
-) -> None:
-    """
-    Remove stale GCP folders and their relationships.
-    """
-    run_cleanup_job("gcp_crm_folder_cleanup.json", neo4j_session, common_job_parameters)
+    load(
+        neo4j_session,
+        GCPFolderSchema(),
+        data,
+        lastupdated=gcp_update_tag,
+        ORG_RESOURCE_NAME=org_resource_name,
+    )
 
 
 @timeit
 def sync_gcp_folders(
     neo4j_session: neo4j.Session,
-    crm_v2: Resource,
     gcp_update_tag: int,
     common_job_parameters: Dict,
-) -> None:
+    org_resource_name: str,
+    defer_cleanup: bool = False,
+) -> List[Dict]:
     """
     Get GCP folder data using the CRM v2 resource object, load the data to Neo4j, and clean up stale nodes.
+    :param org_resource_name: Full organization resource name (e.g., "organizations/123456789012")
+    :param defer_cleanup: If True, skip the cleanup job. Used for hierarchical cleanup scenarios.
```

Comment:
Maybe add a bit more color here to this description. When do we want this to be set true, when do we want this to be set false.